external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -307,14 +317,14 @@ private[kafka010] class KafkaSourceProvider extends DataSourceRegister
     }
   }
 
-  private def validateStreamOptions(caseInsensitiveParams: Map[String, String]) = {
+  private def validateStreamOptions(caseInsensitiveParams: CaseInsensitiveMap[String]) = {
     // Stream specific options
     caseInsensitiveParams.get(ENDING_OFFSETS_OPTION_KEY).map(_ =>
       throw new IllegalArgumentException("ending offset not valid in streaming queries"))
     validateGeneralOptions(caseInsensitiveParams)
   }
 
-  private def validateBatchOptions(caseInsensitiveParams: Map[String, String]) = {
+  private def validateBatchOptions(caseInsensitiveParams: CaseInsensitiveMap[String]) = {
```

Comment:
maybe we can rename the parameter name to `params`, as we can easily tell it's case insensitive by the type.

Reply:
Good simplification, on the way...

Reply:
Renamed on the places where it's not used together with the non case sensitive one + fixed some style problems.